File: Python/stack.py
class Stack():
    def __init__(self):
        self.index = 0
        self.element = []

    def push(self, value):
        self.element.insert(self.index, value)
        self.index += 1

    def pop(self):
        if (self.index == 0):
             return

        self.index -= 1
        result = self.element[self.index]
        del self.element[self.index]
        return result

    def size(self):
        return self.index
    
    def peek(self):
        return self.element[self.index -1]

    def min(self):
        return self.element[0]


# TEST

def main():
    stack = Stack()

    stack.push(5)
    stack.push(6)
    stack.push(7)
    stack.push(8)

    max = stack.peek()
    min = stack.min()

    print(f"\n Max value in stack: {max}")
    print(f"\n Min value in stack: {min}")
    print(f"\n Currente size in stack: {stack.size()}")

    stack.pop()

    print(f"\n Max value in stack: {stack.peek()}")
    print(f"\n Currente size in stack: {stack.size()}")
    return


main()

File: DataStructures/JavaScript/stack.js
var Stack = function(){
    this.index = 0;
    this.element = {};

    // push() method: put value onto the end of the stack
    this.push = function(value){
        this.element[this.index] = value;
        this.index++;
    }

    // pop() method: removes and return the last element of the stack
    this.pop = function(){
        if(this.index === 0) return undefined;

        this.index--;
        var result = this.element[this.index];
        delete this.element[this.index];
        return result;
    }

    // size() method: return the number of the index onto the stack
    this.size = function(){
        return this.index;
    }

    // peek() method: return the last element of the stack
    this.peek = function(){
        return this.element[this.index-1];
    }

}